editor/modules/map/editor-state/index.ts

```diff
@@ -154,6 +153,7 @@ export class EditorState implements MouseHandler, KeyHandler {
         switch (this.state) {
             case State.Palette: next_state = this.palette_area.mouseMovePalette(event); break;
             case State.Editor: next_state = this.editor_area.mouseMoveEditor(event); break;
+            default: throw 'Unreachable code reached';
```

Comment:
Oh nice catch! I think that it's because TypeScript does not type check correctly switch statements when used with enum. It doesn't seems to see when all variant are matched. So without this line I was getting an error on next_state that could have been not initialized after that switch.